images/sds-health-watcher-controller/pkg/controller/sds_infra_watcher.go

```diff
@@ -118,16 +119,39 @@ func RunSdsInfraWatcher(
 				log.Info("[RunSdsInfraWatcher] every LVMVolumeGroup node is in a Running state")
 			}
 
-			if len(missedNodes) == 0 && len(notReadyNodes) == 0 {
-				for _, lvg := range lvgs {
-					log.Debug(fmt.Sprintf("[RunSdsInfraWatcher] tries to update the LVMVolumeGroup %s condition %s to True", lvg.Name, nodeReadyType))
-					err = updateLVGConditionIfNeeded(ctx, cl, log, &lvg, metav1.ConditionTrue, nodeReadyType, "NodesReady", "selected nodes were found in the cluster and have Ready state")
-					if err != nil {
-						log.Error(err, fmt.Sprintf("[RunSdsInfraWatcher] unable to add a condition to the LVMVolumeGroup %s", lvg.Name))
-						continue
+			// Create a list of LVGs that are not in notReadyNodes and missedNodes
+			allProblemNodes := make(map[string]bool)
+			for _, node := range missedNodes {
+				allProblemNodes[node] = true
+			}
+			for _, node := range notReadyNodes {
+				allProblemNodes[node] = true
+			}
+
+			// Find LVGs that should be updated to True status
+			lvgsToUpdate := make([]v1alpha1.LVMVolumeGroup, 0)
+			for _, lvg := range lvgs {
+				shouldUpdate := true
+				for _, node := range lvg.Status.Nodes {
+					if allProblemNodes[node.Name] {
+						shouldUpdate = false
+						break
```

Comment:
We can use empty value in map for set

```suggestion
			// Create a list of LVGs that are not in notReadyNodes and missedNodes
			allProblemNodes := make(map[string]struct{}, len(missedNodes)+len(notReadyNodes))
			for _, node := range missedNodes {
				allProblemNodes[node] = struct{}{}
			}
			for _, node := range notReadyNodes {
				allProblemNodes[node] = struct{}{}
			}

			// Find LVGs that should be updated to True status
			lvgsToUpdate := make([]v1alpha1.LVMVolumeGroup, 0, len(lvgs))
			for _, lvg := range lvgs {
				shouldUpdate := true
				for _, node := range lvg.Status.Nodes {
					if _, found := allProblemNodes[node.Name]; found {
						shouldUpdate = false
						break
```